tools/commentMonitor/main.go

```diff
@@ -63,103 +101,137 @@ func main() {
+				log.Fatalf("matching command not found. comment validation failed")
+			}
+			log.Println("comment validation successful")
+		}
 
-		// Github client for posting comments.
-		clt := newClient(os.Getenv("GITHUB_TOKEN"))
-
-		// Check author association.
-		if *verifyUser {
-			if err := memberValidation(*e.GetComment().AuthorAssociation); err != nil {
+		// Verify if user is allowed to perform activity.
+		if !cmClient.verifyUserDisabled {
+			var allowed bool
+			allowedAssociations := []string{"COLLABORATOR", "MEMBER", "OWNER"}
+			for _, a := range allowedAssociations {
+				if a == authorAssociation {
+					allowed = true
+				}
+			}
+			if !allowed {
 				log.Printf("author is not a member or collaborator")
-				if err := postComment(clt, owner, repo, prnumber, fmt.Sprintf("Error: %s is not a group member nor a collaborator and cannot execute benchmarks.", *e.Sender.Login)); err != nil {
-					log.Fatalln(err)
+				b := fmt.Sprintf("@%s is not a org member nor a collaborator and cannot execute benchmarks.", author)
+				if err := cmClient.ghClient.postComment(ctx, b); err != nil {
+					log.Fatalf("%v : couldn't post comment", err)
 				}
-				os.Exit(78)
+				os.Exit(1)
 			}
-			log.Printf("author is member or collaborator")
-		}
-
-		// Validate comment.
-		args, err := regexValidation(*regex, *e.GetComment().Body)
-		if err != nil {
-			log.Printf("matching command not found")
-			os.Exit(78)
+			log.Println("author is a member or collaborator")
 		}
 
-		// Save args to file. Stores releaseVersion in ARG_0 and prnumber in ARG_1.
-		args = append(args, strconv.Itoa(prnumber))
-		writeArgs(args, *output)
+		// Extract args if regexString provided.
+		if cmClient.regexString != "" {
+			// To have unnamed arguments start at ARG_0 and so forth.
+			unnamedArgCount := 0
+
+			// Add comment arguments.
+			commentArgs := cmClient.regex.FindStringSubmatch(commentBody)[1:]
+			commentArgsNames := cmClient.regex.SubexpNames()[1:]
+			for i, argName := range commentArgsNames {
+				if argName == "" {
+					argName = fmt.Sprintf("ARG_%v", unnamedArgCount)
+					unnamedArgCount++
+				}
+				cmClient.allArgs[argName] = commentArgs[i]
+			}
 
-		envVars := make(map[string]string)
-		for _, e := range os.Environ() {
-			tmp := strings.Split(e, "=")
-			envVars[tmp[0]] = tmp[1]
-		}
+			// Add non-comment arguments if any.
+			cmClient.allArgs["PR_NUMBER"] = strconv.Itoa(prnumber)
 
-		var buf bytes.Buffer
-		commentTemplate := template.Must(template.New("Comment").Parse(os.Getenv(*templateEnvVar)))
-		if err := commentTemplate.Execute(&buf, envVars); err != nil {
-			log.Fatalln(err)
+			// write arguments to fs
+			err := cmClient.writeArgs()
+			if err != nil {
+				log.Fatalf("%v: could not write args to fs", err)
+			}
 		}
 
-		if err := postComment(clt, owner, repo, prnumber, buf.String()); err != nil {
-			log.Fatalln(err)
+		// Post generated comment to Github pr if COMMENT_TEMPLATE is set.
```

Comment:
when an additional comment is required this is usually a sign that it is time to put the next block in its own func.